tomb-common/src/banyan_api/requests/core/auth/device_api_key/regwait.rs

```diff
@@ -0,0 +1,61 @@
+use crate::banyan_api::requests::ApiRequest;
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+use uuid::Uuid;
+
+#[derive(Debug, Serialize, Default)]
+pub struct Regwait {
+    pub nonce: Uuid,
+}
+
+impl Regwait {
+    pub fn new() -> Self {
+        Self {
+            nonce: uuid::Uuid::new_v4(),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RegwaitResponse {
+    pub account_id: Uuid,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RegwaitError {
+    msg: String,
+}
+
+impl std::fmt::Display for RegwaitError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.msg)
+    }
+}
+
+impl Error for RegwaitError {}
+
+impl ApiRequest for Regwait {
+    type ResponseType = RegwaitResponse;
+    type ErrorType = RegwaitError;
+
+    fn build_request(
+        self,
+        base_url: &reqwest::Url,
+        client: &reqwest::Client,
+    ) -> reqwest::RequestBuilder {
+        // Create the full url
+        let full_url = base_url
+            .join(&format!(
+                "/api/v1/auth/device_api_key/start_regwait/{}",
+                self.nonce
+            ))
+            .unwrap();
+        // Run a get request
+        client.get(full_url).json(&self)
+    }
+
+    fn requires_authentication(&self) -> bool {
+        // No auth required
+        false
+    }
+}
```

Comment:
It occurred to me that we have a value we can use to identify these requests better and more simply than a random nonce that should work with the existing registration and can likely be worked around with the existing functionality (though the frontend will still have to play a little bit of a role).

If we switch from a nonce to the fingerprint of the key being registered, we don't need to generate or share an extra value between the client and server. It can be computed directly on either side. The frontend currently puts the key directly in the database and would need to make an extra HTTP call to the core service (authenticated using the browser's API device key credentials) so that a signal can be sent to this waiting connection... But that is a much smaller lift than moving everything over and gets this working much quicker.

Reply:
noted; will implement